hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/container/upgrade/UpgradeManager.java

```diff
@@ -0,0 +1,206 @@
+    final DatanodeDetails detail =
+        UpgradeUtils.getDatanodeDetails(configuration);
+
+    final MutableVolumeSet dataVolumeSet = UpgradeUtils
+        .getHddsVolumes(configuration, StorageVolume.VolumeType.DATA_VOLUME,
+            detail.getUuidString());
+
+    HddsVolumeUtil.loadAllHddsVolumeDbStore(dataVolumeSet, null, false, LOG);
+    initVolumeStoreMap(dataVolumeSet, configuration);
+    upgradeAll(dataVolumeSet, configuration);
+  }
+
+  public void initVolumeStoreMap(MutableVolumeSet dataVolumeSet,
+      OzoneConfiguration configuration) throws IOException {
+    for (StorageVolume storageVolume : dataVolumeSet.getVolumesList()) {
+      final HddsVolume volume = (HddsVolume) storageVolume;
+      final File containerDBPath = getContainerDBPath(volume);
+      final DatanodeStoreSchemaThreeImpl datanodeStoreSchemaThree =
+          new DatanodeStoreSchemaThreeImpl(configuration,
+              containerDBPath.getAbsolutePath(), false);
+      volumeStoreMap.put(volume.getStorageDir().getAbsolutePath(),
+          datanodeStoreSchemaThree);
+    }
+  }
+
+  public List<Result> upgradeAll(MutableVolumeSet volumeSet,
+                                 OzoneConfiguration configuration) {
+    List<Result> results = new ArrayList<>();
+    final List<StorageVolume> volumesList = volumeSet.getVolumesList();
+    Map<HddsVolume, CompletableFuture<Result>> volumeFutures =
+        new HashMap<>();
+    long startTime = System.currentTimeMillis();
+
+    LOG.info("Start upgrade {} volumes container LayoutVersion",
+        volumesList.size());
+
+    for (StorageVolume volume : volumesList) {
+      final HddsVolume hddsVolume = (HddsVolume) volume;
+      final UpgradeTask task =
+          new UpgradeTask(configuration, hddsVolume,
+              getDBStore(hddsVolume));
+
+      final CompletableFuture<Result> future =
+          task.getUpgradeFutureByVolume();
+      volumeFutures.put(hddsVolume, future);
+    }
+
+    for (Map.Entry<HddsVolume, CompletableFuture<Result>> entry : volumeFutures
+        .entrySet()) {
+      final HddsVolume hddsVolume = entry.getKey();
+      final CompletableFuture<Result> volumeFuture = entry.getValue();
+
+      try {
+        final Result result = volumeFuture.get();
+        results.add(result);
+        LOG.info("Finish upgrade containers on volume {}, result {}",
+            hddsVolume.getVolumeRootDir(), result);
+      } catch (Exception e) {
+        LOG.error("Upgrade containers on volume {} failed",
+            hddsVolume.getVolumeRootDir(), e);
+      }
+    }
+
+    LOG.info("Upgrade all volume container LayoutVersion costs {}s",
+        (System.currentTimeMillis() - startTime) / 1000);
+    return results;
+  }
+
+  @VisibleForTesting
+  public DatanodeStoreSchemaThreeImpl getDBStore(HddsVolume volume) {
+    return volumeStoreMap.get(volume.getStorageDir().getAbsolutePath());
+  }
+
+  /**
+   * This class response upgrade v2 to v3 container result.
```

Comment:
This class response upgrade v2 to v3 container result. ->

This class contains v2 to v3 container upgrade result.